### PR TITLE
Development Documentation

### PR DESCRIPTION
**Pull request by Phocacius**
For mapbender 3, documentation for development has been spread across multiple locations and varied greatly in precision and up-to-dateness.

This PR aims on bundling all developer documentation into a single directory living in the mapbender core directory and therefore automatically versioned.

### CODE REVIEW

File: docs/controllers/controllers.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+ Regex        #^
+                 /user/login
+             $#x
+```
+
+To learn more about routing, read the [Symfony documentation](https://symfony.com/doc/current/book/index.html).
+
+### Defining routes using annotations
+
+In Mapbender, we use decentralized route definitions: Instead of writing each and every route in the routing.yml, we import their definition from the controller classes in the activated bundles. This has the advantage of having the definition with the controller function. This should usually be fine and can be - if need arises - easily overwritten by adapting the routing.yml.
```
Comment (astroidex) [inline docs/controllers/controllers.md:64]: 2x in text rounting.yaml

File: docs/controllers/controllers.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+* **ApplicationController**: Manage applications
+* **GroupController**: Manage user groups
+* **LayerController**: Manage layers
+* **RepositoryController**: Manage the layer repository
+* **SettingsController**: Manage common settings
+* **UserController**: Manage users
+* **ManagerController**: Provides common functionality for the other controllers
+
+Each of the these controllers (right now work is going on within the ApplicationController) is a good example of what we think of as good kinda RESTful URLs.
```
Comment (astroidex) [inline docs/controllers/controllers.md:102]: kinda -> ?

File: docs/elements/elements.md [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+
+```
+
+The passive way for communication is to subscribe to events of another target. You also need to know the HTML element,
+but you can now listen for the other widget to call your widget. This is done using default jQuery events. <!--Well, almost
+default jQuery events: If you use the "_trigger" method provided by the jQuery UI widget factory-->
+
+## How to create your own Element?
+
+Mapbender offers an app/console command to create different elements:
```
Comment (astroidex) [inline docs/elements/elements.md:124]: bin/console 
Is the command to create an element still working? Else delete this section

File: docs/getting_started/conventions.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
+# Conventions for Mapbender
```
Comment (astroidex) [inline docs/getting_started/conventions.md:1]: I guess we have to discuss the conventions

File: docs/getting_started/faq.md [hunk 5]
```diff
@@ -0,0 +1,156 @@
+# Frequently Asked Questions
```
Comment (astroidex) [inline docs/getting_started/faq.md:1]: Is this page necessary in the developer docu. We have https://docs.mapbender.org/develop/en/faq.html

File: docs/workflows/translation.md [hunk 6]
```diff
@@ -0,0 +1,224 @@
+# Translation
```
Comment (astroidex) [inline docs/workflows/translation.md:1]: Maybe this page should stay in the normal documentation and refer to it.
- https://docs.mapbender.org/develop/en/architecture/translation.html
the developer docu could containt some extended info like the bin/console command to generate the translation

**Review by astroidex — CHANGES_REQUESTED** [review]
- we should discuss what to publish with version 4
- some parts can come in future releases
- would be good to have the developer docu somehow published online too

File: docs/controllers/controllers.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+The mapping from request path to controller function is basically done in the configuration, where the routing.yml defines these mappings - called routes - or imports their definitions from bundles (or other files).
+In your own bundle, you can also use [PHP attributes directly in your controller](https://symfony.com/doc/current/routing.html).
+
+To get an overview of all defined routes by using the console command.
+
+```console
+ cd mapbender/application
+ app/console router:debug
+
+ [router] Current routes
```
[the remark this answers is not in the document]
Comment (Phocacius) [reply docs/controllers/controllers.md:18]: it doesn't really matter, it's just an example anyway

File: docs/getting_started/conventions.md (hunk 4 above)
Comment (Phocacius) [reply docs/getting_started/conventions.md:1]: will be done for the next release

File: docs/getting_started/faq.md (hunk 5 above)
Comment (Phocacius) [reply docs/getting_started/faq.md:1]: it's different content. Will be refactoed / unified in a future release.

**Pull request merged**